Update "official marketplace" copy

KokkieH opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->
We're updating the "Official WooCommerce Marketplace" copy for the in-app marketplace to just "The WooCommerce Marketplace" to avoid any potential confusion with the "Official" badge created for payment partners.
<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

* Update all instances of "Official WooCommerce Marketplace" copy, and associated tests:
    - Marketplace header
    - Product preview modal
    - Marketplace links in onboarding checklist tasks
* Update additional payment options link in new payment settings to point to in-app Marketplace
    - This is how the links worked in the old payment recommendation elements. New payment settings introduced a regression by going back to using an external WCCOM URL.
    - Added tests for both the link navigation and the Tracks event
* Add fallback to WooCommerce.com for all marketplace CTAs if in-app marketplace is disabled.
<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->

Closes [WCCOM-1420](https://linear.app/a8c/issue/WCCOM-1420/remove-official-terminology)

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:
1. Check out this branch and apply [this patch](https://github.com/user-attachments/files/20628758/tasklistforce.patch) that will force all relevant task list items to show. Also make sure your store location is set to the US.

2. In the onboarding checklist, click Customize your store. Confirm "Official" is no longer in the Marketplace CTA card title, and that clicking it still takes you to the in-app Extensions page on the Themes tab.
![Home_‹_WooCommerce_‹_woocommerce_—_WooCommerce](https://github.com/user-attachments/assets/a1b2a8e7-babe-4818-921b-2ee0f0bdb0ea)

3. In the onboarding checklist, click Add your products. Confirm the marketplace CTA copy is updated, and that clicking it still takes you to the in-app Extensions page with the Merchandising category selected.
![Home_‹_WooCommerce_‹_woocommerce_—_WooCommerce](https://github.com/user-attachments/assets/268abae0-220e-414b-8433-25f372240722)

4. In the onboarding checklist, click the Collect Sales Tax task. Confirm the marketplace CTA copy is updated and that clicking it still takes you to the in-app Extensions page with the Operations category selected.
![Home_‹_WooCommerce_‹_woocommerce_—_WooCommerce](https://github.com/user-attachments/assets/ba6362c4-e5bc-42a7-8e49-af55a96dc6d7)

5. In the onboarding checklist, click the Select your shipping options task. Confirm the marketplace CTA copy is updated, and that clicking it still takes you to the in-app Extensions page with the Shipping category selected.
![Home_‹_WooCommerce_‹_woocommerce_—_WooCommerce](https://github.com/user-attachments/assets/de239c7c-87fe-4dca-8be2-d37684992819)

6. Go to WooCommerce->Settings->Shipping. If you see a Woo Shipping card, confirm the marketplace CTA copy is updated and link behaves same as the one in step 5.
![WooCommerce_settings_‹_woocommerce_—_WordPress](https://github.com/user-attachments/assets/d37f913d-0ec3-48a6-88d8-4f7b108f6672)

7. Next click the Get Started button. Else visit http://localhost:8888/wp-admin/admin.php?page=wc-admin&task=shipping-recommendation directly. The CTA should be updated as per step 5

8. Go to WooCommerce->Settings->Payment. Expand the Other Payment Options card, and confirm the marketplace CTA at the bottom is consistent with the other marketplace CTAs above
![WooCommerce_settings_‹_woocommerce_—_WordPress](https://github.com/user-attachments/assets/9b80845d-5f0a-4a27-8a43-224a220bcac6)

9. Run `localStorage.setItem( 'debug', 'wc-admin:*' );` in the console and persist the logs. This will log tracks events to the console for you to see (If you're using Chrome, enable verbose messages in the console.). Click the link. Confirm that you're taken to the Extensions screen on the Payment Gateways category, and that the `wcadmin_settings_payments_recommendations_other_options` event is fired.
![WooCommerce_settings_‹_woocommerce_—_WordPress](https://github.com/user-attachments/assets/d44c8ae9-c27f-486e-a7b0-6809d5c8a061)

10. Go to the `/plugins/woocommerce/client/admin` directory, and run the following tests to make sure everything still passes following the changes:
```
pnpm test:js client/payments/test/payment-recommendations.test.tsx
pnpm test:js client/task-lists/fills/products/test/index.tsx
pnpm test:js client/shipping/test/experimental-shipping-recommendations.tsx
pnpm test:js client/task-lists/fills/tax/test/index.tsx
pnpm test:js client/task-lists/fills/experimental-shipping-recommendation/test/shipping-recommendation.tsx
pnpm test:js client/task-lists/fills/shipping/test/index.tsx
pnpm test:js client/task-lists/fills/PaymentGatewaySuggestions/test/index.js
pnpm test:js client/settings-payments/test/settings-payments-main.test.tsx
```

11. Add this snippet to the Code Snippets plugin (or `functions.php`):
```
add_filter( 'pre_option_woocommerce_feature_marketplace_enabled',function ( $enabled ) { return 'no'; } );
```

12. Check each of the above steps again. The CTA links should now go to the correct category listing on WooCommerce.com. Also confirm the various Tracks events for clicking the CTA are also still recorded.

<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->
* Update all instances of "Official WooCommerce Marketplace" copy, and associated tests:
    - Marketplace header
    - Product preview modal
    - Marketplace links in onboarding checklist tasks
* Update additional payment options link in new payment settings to point to in-app Marketplace
    - This is how the links worked in the old payment recommendation elements. New payment settings introduced a regression by going back to using an external WCCOM URL.
    - Added tests for both the link navigation and the Tracks event
* Add fallback links to WooCommerce.com for when marketplace features are disabled
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

KokkieH committed:
Update all instances of "Official WooCommerce Marketplace" copy

- Update marketplace links and copy in task list items

- Update Extensions page header

- Update product preview modal header

KokkieH committed:
Update Marketplace copy on Customize your store page

KokkieH committed:
Update marketplace link in payments settings to link to in-app marketplace

- New payments settings introduced a regression from how this link behaved in the old payment recommendations

- Add tests for marketplace link

KokkieH committed:
Merge branch 'trunk' into update/WCCOM-1420-official-marketplace-copy

KokkieH committed:
fix linting

Commit:
Add changefile(s) from automation for the following project(s): woocommerce, woocommerce/client/admin

vladolaru commented on plugins/woocommerce/client/admin/client/settings-payments/settings-payments-main.tsx:
```diff
@@ -447,16 +448,17 @@ export const SettingsPaymentsMain = () => {
+		<TrackedLink
+			message={ __(
+				// translators: {{Link}} is a placeholder for a html element.
+				'Visit {{Link}}The WooCommerce Marketplace{{/Link}} to find additional payment providers.',
```
Shouldn't this be lowercase t? Also, shouldn't we keep up with the section's "Other payment options" naming and say "find additional payment options"?

```suggestion
				'Visit {{Link}}the WooCommerce Marketplace{{/Link}} to find additional payment options.',
```

KokkieH replied:
@pmcpinto in the Linear task you provided the new copy as `The WooCommerce Marketplace`. Is the capital `T` an intentional choice?

Whether capitalized or not, I think we should be consistent across all the different placements of this CTA. At the moment, they all use a capital `T`.

KokkieH replied:
Updated in ef6f56a

vladolaru committed:
Merge branch 'trunk' into update/WCCOM-1420-official-marketplace-copy

vladolaru commented on plugins/woocommerce/client/admin/client/settings-payments/settings-payments-main.tsx:
```diff
@@ -447,16 +448,17 @@ export const SettingsPaymentsMain = () => {
+				'woocommerce'
+			) }
+			onClickCallback={ trackMorePaymentsOptionsClicked }
+			targetUrl={ getAdminLink(
```
The Marketplace is a behind a (hidden) feature flag. While the UI may not be available, all it takes is this snippet to disable it:
```
add_filter( 'pre_option_woocommerce_feature_marketplace_enabled',function ( $enabled ) { return 'no'; } );
```
If the feature is disabled, we should fallback on sending folks to woocommerce.com. Something like this would do it:

```
targetUrl={
				isFeatureEnabled( 'marketplace' )
					? getAdminLink(
							'admin.php?page=wc-admin&tab=extensions&path=/extensions&category=payment-gateways'
					  )
					: 'https://woocommerce.com/product-category/woocommerce-extensions/payment-gateways/'
			}
```

KokkieH replied:
Thanks, @vladolaru. I wasn't aware the marketplace could be disabled so easily.

In that case, we'd need to add this fallback to the other placements as well. cc @pmcpinto

KokkieH replied:
Fallback to WCCOM added in 3a3f9fa for all the marketplace CTAs

vladolaru reviewed (commented):
Thanks for the updates, @KokkieH. I've checked and from a Payments Settings page perspective it works.

However I think we need to provide a fallback to woocommerce.com for all those stores that get their `marketplace` feature disabled (either by a host, an agency, or by some optimization plugin).

KokkieH committed:
Merge branch 'trunk' into update/WCCOM-1420-official-marketplace-copy

KokkieH committed:
Add fallback marketplace link to WooCommerce.com if marketplace is disabled on the site

KokkieH committed:
Add isFeatureEnabled mock to marketplace CTA tests

KokkieH committed:
Add fallback link for Marketing page CTA

KokkieH committed:
Merge branch 'trunk' into update/WCCOM-1420-official-marketplace-copy

KokkieH committed:
Fix PaymentRecommendations component fallback link and test

KokkieH committed:
Update use of isFeatureEnabled mock so it does not throw a lint error

KokkieH committed:
Use lowercase t for CTA links

KokkieH committed:
Fix linting error

vladolaru reviewed (approved):
Thank you for improving the code, @KokkieH! I tested the Payments Settings link and it works as expected, both with and without the `marketplace` feature enabled. Approving it!

gedex commented:
Also, not sure if we'd like to update copy in `readme.txt`:

https://github.com/woocommerce/woocommerce/blob/45c11368fc4a65c4e48aa18037a121ec776cc81d/plugins/woocommerce/readme.txt#L43

gedex reviewed (approved):
Checked each file change and it looks good. Tested followings and works as expected. Thanks for updating it!

* [x] Customize your theme
* [x] Add your products
* [x] Collect sales tax
* [x] Select your shipping options
* [x] WooCommerce > Settings > Shipping
* [x] Get your products shipped
* [x] Other payment options

KokkieH committed:
Update "Official" terminology in WC README file

KokkieH commented:
> With a latest trunk, the copy is different:

Correct. In Feb we updated all instances of that copy to the new "Official Marketplace" version, but the new payment settings reverted it back to the old version. This PR includes updating that to match our other Marketplace CTAs in onboarding and settings.

> Also, not sure if we'd like to update copy in readme.txt:

Good catch! Updated in a6d8a81bc8

KokkieH committed:
Merge branch 'trunk' into update/WCCOM-1420-official-marketplace-copy

Commit:
Add changefile(s) from automation for the following project(s): woocommerce, woocommerce/client/admin

KokkieH committed:
Merge branch 'trunk' into update/WCCOM-1420-official-marketplace-copy
